### PR TITLE
[SPARK-24886][INFRA] Fix the testing script to increase timeout for Jenkins build (from 300m to 340m)

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

Currently, looks we hit the time limit time to time. Looks better increasing the time a bit.

For instance, please see https://github.com/apache/spark/pull/21822

For clarification, current Jenkins timeout is 400m. This PR just proposes to fix the test script to increase it correspondingly.

*This PR does not target to change the build configuration*

## How was this patch tested?

Jenkins tests.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @rxin

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by hvanhovell** [comment]
@HyukjinKwon do we have any idea why we are hitting a timeout?

**Comment by HyukjinKwon** [comment]
I am not really sure on that. I asked the same question before and got no answer before. Just vaguely roughly guess there's something wrong in Jenkins cluster - I have roughly been kind of keen to check build time and to me seems suddenly increased (in some cases or some machines(?)).

**Comment by HyukjinKwon** [comment]
Just given observation for the builds in #21822, most of timeouts looked happened in `amp-jenkins-worker-06` machine FWIW.

**Comment by rxin** [comment]
This helps, but it is not sustainable to keep increasing the threshold. What we need to do is to look at test time distribution and figure out what test suites are unnecessarily long and actually cut down the time there. @HyukjinKwon  Would you be interested in doing that?

**Comment by HyukjinKwon** [comment]
of course i am as usual. I actually already have been being taking care of it. Thing is the tests are just being added even if they are duplicated of something. I feel like it's a bit excessive so far. In genetal, I don't think there are particular tests especially taking a lot of time IMHO. What we should do is that we put some efforts to deduplicate the tests.

**Comment by rxin** [comment]
Are more pull requests failing due to time out right now?

**Comment by HyukjinKwon** [comment]
Yup, looks so in your PR https://github.com/apache/spark/pull/21822#issuecomment-407298841

**Comment by rxin** [comment]
If that's the only one I think that PR itself needs to be fixed
(significantly increases test runtime), and I wouldn't increase the time
here.

**Comment by dilipbiswal** [comment]
@HyukjinKwon I saw the following test run for 11 minutes on jenkins for one of my PR. Not sure if its a transient problem. Just thought, i should let you know. On the nightly runs, should we have test that runs for that long ?

SPARK-22499: Least and greatest should not generate codes beyond 64KB (11 minutes, 38 seconds)

**Comment by HyukjinKwon** [comment]
Ah, of course we shouldn't. I roughly checked few builds in https://github.com/apache/spark/pull/21822 and made a fix for the test https://github.com/apache/spark/pull/21855

**Comment by dilipbiswal** [comment]
@HyukjinKwon Super. Thanks a lot for fixing.

**Comment by HyukjinKwon** [comment]
I am reopening this per https://github.com/apache/spark/pull/21898#issuecomment-410909703

cc @cloud-fan, @rxin and @shaneknapp

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
I think we still intermediately meet this limit issue. For instance:

https://github.com/apache/spark/pull/22001#issuecomment-410809387
https://github.com/apache/spark/pull/21923#issuecomment-409072530
https://github.com/apache/spark/pull/21909#issuecomment-410380074
https://github.com/apache/spark/pull/21777#issuecomment-405153959
https://github.com/apache/spark/pull/21156#issuecomment-403571839
https://github.com/apache/spark/pull/21109#issuecomment-382957092
https://github.com/apache/spark/pull/20838#issuecomment-405764177
https://github.com/apache/spark/pull/20345#issuecomment-359670479

I saw here multiple times - https://github.com/apache/spark/pull/21991#issuecomment-410398112

https://github.com/apache/spark/pull/21991#issuecomment-410398112
https://github.com/apache/spark/pull/21930#issuecomment-409802056
https://github.com/apache/spark/pull/21972#issuecomment-410067354

**Comment by shaneknapp** [comment]
i'm also more than happy to bump the timeout in the PRB build, but i think that's just putting duct tape on a band-aid and spray painting it to hide the layers of tape.

the builds and tests just take too long.  i know that solving this problem is far beyond the scope of this PR, but build duration really needs some attention.

File: dev/run-tests-jenkins.py [hunk 1]
```diff
@@ -181,8 +181,8 @@ def main():
     short_commit_hash = ghprb_actual_commit[0:7]
 
     # format: http://linux.die.net/man/1/timeout
-    # must be less than the timeout configured on Jenkins (currently 350m)
-    tests_timeout = "300m"
+    # must be less than the timeout configured on Jenkins (currently 400m)
```
Comment (HyukjinKwon) [inline dev/run-tests-jenkins.py]: Updated by http://apache-spark-developers-list.1001551.n3.nabble.com/build-system-bumped-pull-request-builder-job-timeout-to-400mins-td24647.html

**Comment by HyukjinKwon** [comment]
Let me push this in late tonight or early tomorrow.

**Comment by HyukjinKwon** [comment]
I am getting this in. We are seeing more - https://github.com/apache/spark/pull/22011#issuecomment-411918221

**Comment by HyukjinKwon** [comment]
Merged to master.

File: dev/run-tests-jenkins.py [hunk 2]
```diff
@@ -181,8 +181,8 @@ def main():
...
+    tests_timeout = "340m"
```
Comment (shaneknapp) [inline dev/run-tests-jenkins.py]: we're STILL seeing test timeouts.  let's bump this to 400m and i'll up the timeout in jenkins to 430m.

Comment (HyukjinKwon) [reply dev/run-tests-jenkins.py]: Woah .. got it ..

**Pull request closed without merging**